packages/ember-data/lib/system/model/model.js

```diff
@@ -844,6 +817,35 @@ var Model = Ember.Object.extend(Ember.Evented, {
   },
 
   /**
+    @method _changedKeys
+    @private
+  */
+  _changedKeys: function(updates) {
+    var changedKeys = [];
+
+    if (updates && typeof updates === 'object') {
+      var original, i, value, key;
+      var keys = Ember.keys(updates);
+      var length = keys.length;
+
+      original = merge({}, this._data);
+      original = merge(original, this._attributes);
```

Comment:
This is required for the scenario in test https://github.com/emberjs/data/pull/2998/files#diff-56dbbdcd2b2c38cfe44c52bec2b99646R55